src/Orleans/Utils/DictionaryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans
+{
+    static class DictionaryExtensions
+    {
+        public static TResult Find<TKey, TResult>(this IDictionary<TKey, TResult> dictionary, TKey key) where TResult : class
```

Comment:
Why? What's the point of adding documentation for completely obvious parts of codebase, which is also internal? This is not Java :) ...
`IReadOnlyDictionary` ?